[TensorExpr] Move 'initializer' field from 'Tensor' to 'Buf'.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50995 [TensorExpr] Redesign `Tensor` class.
* #50994 [TensorExpr] Move 'lowerToStmt' method from 'LoopNest' to 'Tensor'.
* **#50993 [TensorExpr] Move 'initializer' field from 'Tensor' to 'Buf'.**

This is the first step to make 'Tensor` a thin wrapper over 'Buf' and
'Stmt', which will be finished in subsequent PRs. This change also
allows to remove 'buf_initializers_' from 'LoopNest', making it "less
stateful".

Differential Revision: [D26038224](https://our.internmc.facebook.com/intern/diff/D26038224)

Commit:
[TensorExpr] Move 'initializer' field from 'Tensor' to 'Buf'.

This is the first step to make 'Tensor` a thin wrapper over 'Buf' and
'Stmt', which will be finished in subsequent PRs. This change also
allows to remove 'buf_initializers_' from 'LoopNest', making it "less
stateful".

[ghstack-poisoned]

bertmaher reviewed (approved):
I'm kind of ambivalent here because it feels like initialization is part of "compute" rather than of the data.  But it also doesn't feel like it belongs on tensor either, so... surewhynot.gif

Commit:
Update on "[TensorExpr] Move 'initializer' field from 'Tensor' to 'Buf'."


This is the first step to make 'Tensor` a thin wrapper over 'Buf' and
'Stmt', which will be finished in subsequent PRs. This change also
allows to remove 'buf_initializers_' from 'LoopNest', making it "less
stateful".

Differential Revision: [D26038224](https://our.internmc.facebook.com/intern/diff/D26038224)

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Move 'initializer' field from 'Tensor' to 'Buf'."


This is the first step to make 'Tensor` a thin wrapper over 'Buf' and
'Stmt', which will be finished in subsequent PRs. This change also
allows to remove 'buf_initializers_' from 'LoopNest', making it "less
stateful".

Differential Revision: [D26038224](https://our.internmc.facebook.com/intern/diff/D26038224)

[ghstack-poisoned]

navahgar reviewed (approved):
LGTM
